person.rb

```diff
@@ -0,0 +1,23 @@
+class Person
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(age, name = 'unkown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  def can_use_services?
+    true if parent_permission
+    false
+  end
+
+  private
+
+  def of_age?
+    true if age >= 18
+    false
+  end
+end
```

Comment:
- [ ] parent permission should have been declared with "@" at the beginning as it is an instance variable that we declared in the initialize method otherwise it throws an error.
- [ ] You can use the "of_age?" method inside the "can_use_services?" method to not repeat yourself in different methods as follow: 

```
def can_use_services? 
    is_of_age?() || @parent_permission
end
```